akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -509,6 +512,25 @@ private[impl] final class AgentImpl[A <: Agent](
           new SpiAgent.ModelSettings(p.connectionTimeout().toScala, p.responseTimeout().toScala, p.maxRetries()))
       case p: ModelProvider.Custom =>
         new SpiAgent.ModelProvider.Custom(() => p.createChatModel(), () => p.createStreamingChatModel())
+      case p: ModelProvider.Bedrock =>
+        new SpiAgent.ModelProvider.Bedrock(
+          region = p.region,
+          modelId = p.modelId,
+          logRequests = p.logRequests,
+          logResponses = p.logResponses,
+          returnThinking = p.returnThinking,
+          sendThinking = p.sendThinking,
+          maxOutputTokens = p.maxOutputTokens,
+          reasoningTokenBudget = p.reasoningTokenBudget,
+          additionalModelRequestFields = p.additionalModelRequestFields.asScala.toMap,
+          accessToken = p.accessToken,
+          temperature = p.temperature,
+          topP = p.topP,
+          maxTokens = p.maxTokens,
+          modelSettings = new SpiAgent.ModelSettings(
+            FiniteDuration.apply(30, TimeUnit.SECONDS),
```

Comment:
is this not used?

Reply:
No. There is only single api to set timeouts: `builder.timeout()`